Android/src/org/droidplanner/android/maps/providers/baidu_map/BaiduMapFragment.java

```diff
@@ -0,0 +1,928 @@
+
+        mBDLocClient = new LocationClient(getActivity().getApplicationContext());
+		mBDLocClient.registerLocationListener(mBDLocListener);
+
+        UiSettings mUiSettings = map.getUiSettings();
+        mUiSettings.setCompassEnabled(false);
+        mUiSettings.setOverlookingGesturesEnabled(false);
+        mUiSettings.setZoomGesturesEnabled(false);
+    }
+
+    private void setupMapOverlay(BaiduMap map) {
+        final BaiduMapPrefFragment provider = (BaiduMapPrefFragment)(getProvider().getMapProviderPreferences());
+        final Activity activity = getActivity();
+        final Context context = activity.getApplicationContext();
+
+        int mapType =provider.getMapType(context);
+        map.setMapType(mapType);
+    }
+
+    private void setupOnlineMapOverlay(BaiduMap map) {
+        // not used in baidumap
+        return;
+    }
+
+    private void setupOfflineMapOverlay(BaiduMap map) {
+        // not used in baidumap
+		return;
+	}
+
+    protected void clearMap() {
+		final BaiduMap map = getBaiduMap();
+		if (map != null) {
+		    map.clear();
+            setupMapOverlay(map);
+		}
+    }
+
+    private LatLngBounds getBounds(List<LatLng> pointsList) {
+        LatLngBounds.Builder builder = new LatLngBounds.Builder();
+        for (LatLng point : pointsList) {
+            builder.include(point);
+        }
+        return builder.build();
+    }
+
+    public double getMapRotation() {
+        final BaiduMap map = getBaiduMap();
+        if (map == null) return 0;
+
+		return map.getMapStatus().rotate;
+    }
+
+    public VisibleMapArea getVisibleMapArea(){
+        return null;
+    /*      FIXME: baidu map project doesn't have visible region
+        final BaiduMap map = getBaiduMap();
+        if(map == null)
+            return null;
+
+        final VisibleRegion mapRegion = map.getProjection().getVisibleRegion();
+        return new VisibleMapArea(DroneHelper.LatLngToCoord(mapRegion.farLeft),
+                DroneHelper.LatLngToCoord(mapRegion.nearLeft),
+                DroneHelper.LatLngToCoord(mapRegion.nearRight),
+                DroneHelper.LatLngToCoord(mapRegion.farRight));
+    */
+    }
+
+    @Override
+    public void skipMarkerClickEvents(boolean skip) {
+        useMarkerClickAsMapClick = skip;
+    }
+
+    @Override
+    public void updateRealTimeFootprint(FootPrint footprint) {
+        List<LatLong> pathPoints = footprint == null
+                ? Collections.<LatLong>emptyList()
+                : footprint.getVertexInGlobalFrame();
+
+        if (pathPoints.isEmpty()) {
+            if (mFootprintPoly != null) {
+                mFootprintPoly.remove();
+                mFootprintPoly = null;
+            }
+        } else {
+            if (mFootprintPoly == null) {
+                PolygonOptions pathOptions = new PolygonOptions()
+					    .stroke(new Stroke(FOOTPRINT_DEFAULT_WIDTH, FOOTPRINT_DEFAULT_COLOR))
```

Comment:
@droneboost let's store `new Stroke(...)` into a constant since it's unlikely to change.